### DISCUSSION TITLE
CardDav

### DISCUSSION THREAD

**Discussion by gavdevs** [forwardemail/forwardemail.net#274] [Ideas]
Hey there, I am looking to start using this service! I noticed quite recently that CalDav was added as a feature. I'd love to know if CardDav is something thats planned. CardDav is the only thing right now keeping me from fully committing. (Currently in the process of de-appleing my life and am trying to setup some alternatives to icloud and this hits the box except for contacts storage.

**Comment by titanism** [comment]
@gavyncaldwell we don't support CardDAV yet (it is 100% planned for), but we'll see what we can do in terms of expediting that for you.

**Comment by jandroav** [comment]
Hey! Very interested as well! 🔥

**Comment by ceonelson** [comment]
Yes would love this feature as well!

**Comment by ivanwmco** [comment]
Nice! I have completely de-appled by donating my MacBook Air, AirPods Pro & all accesories to my very grateful 18yo niece and selling my iPhone. I got a Google Pixel phone & tablet which I flashed with GrapheneOS and NitroPad laptop with Ubuntu + Nitrokey 3C NFC. Now I am trying to de-google as much as possible: Cheogram + JMP Chat & Silent Link, Thunderbird Desktop & Mobile + Etar & DAVx5, personalDNSfilter + Invizible Pro & Rethink, etc.

It turns out FE is very well positioned for me to replace Gmail + SimpleLogin/AmonAddy, Google Calendar & Contacts. I have FE email aliases and CalDAV setup and I can't wait to add CardDAV!

**Reply by titanism** [reply]
Founder here, it's a priority this year for sure, will keep you all apprised

**Comment by titanism** [comment]
We've ironed out CalDAV fairly well as of today (see <https://github.com/forwardemail/forwardemail.net/issues/325#issuecomment-2741903094>) and have a firm understanding of DAV specification.  We also have done some R&D on CardDAV, and this may be released sooner than we imagined. 🙏 😓

**Comment by titanism** [comment]
I'm trying to wrap up the tests and implementation.  Here's a work in progress commit: <https://github.com/forwardemail/forwardemail.net/commit/28aaab077a4a7a23ac57ce50d93f7808ca546ecd>

Sorting out routing then writing more tests and going to test in carddav clients.  Will announce a beta soon, then public release.

**Comment by titanism** [comment]
Some great news... CardDAV support is now ready, tested, and all the code is in the repo <https://github.com/forwardemail/forwardemail.net>.  It's fully RFC spec compliant as we can see too, including advanced filtering, etc.  We're spinning up a server and will release it probably today.

**Comment by titanism** [comment]
🎉 @forwardemail's highly demanded CardDAV support is live! 🚀

✅ Quantum-resistant encryption
✅ Privacy-first approach
✅ 100% open-source
✅ No third-party services

🔄 Sync contacts with @Apple, DAVx⁵ (@Android), @mozthunderbird, @emclient, and more!

𝕏 post at <https://x.com/fwdemail/status/1933335889582621002>

Docs at <https://forwardemail.net/faq#do-you-support-contacts-carddav>

![smm_carddav_sync_animation](https://github.com/user-attachments/assets/6bc3162b-628e-4f99-98a4-58cd40269be9)

**Reply by laontme** [reply]
> DAVx⁵

I was trying out your CalDAV and CardDAV implementation with DAVx5 and there's one thing that bothers me, you need to set up two accounts, one for contacts and other one for calendar because of different hosts for carddav and caldav, but it seems you might overcome this by just adding .well-known directly for forwardemail.net with redirect to corresponding host

https://manual.davx5.com/accounts_collections.html#how-does-service-discovery-work
